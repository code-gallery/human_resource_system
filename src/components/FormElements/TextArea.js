import React from 'react'
import PropTypes from 'prop-types'

const TextArea = (props) => {
  const { name, label, onChange } = props
  const defaultValue = props.defaultValue || ''
  return (
    <div className="form-group">
      <label>{label}</label>
      <textarea
        name={name}
        className="form-textarea"
        defaultValue={defaultValue}
        onChange={onChange}
      />
    </div>
  )
}

TextArea.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  defaultValue: PropTypes.string,
  onChange: PropTypes.func.isRequired
}

export default TextArea
