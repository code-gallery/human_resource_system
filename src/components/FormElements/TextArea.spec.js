import React from 'react'
import renderer from 'react-test-renderer'
import TextArea from './TextArea'

describe('<TextArea />', () => {
  const commonProps = {
    name: 'about',
    label: 'About',
    onChange: jest.fn()
  }

  describe('when props.defaultValue is undefined', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <TextArea {...commonProps} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('when props.defaultValue is valid', () => {
    it('renders correctly', () => {
      const props = {
        ...commonProps,
        defaultValue: 'lorem ipsum'
      }
      const tree = renderer.create(
        <TextArea {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
