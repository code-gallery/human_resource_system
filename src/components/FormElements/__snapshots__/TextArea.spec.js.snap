// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> when props.defaultValue is undefined renders correctly 1`] = `
<div
  className="form-group"
>
  <label>
    About
  </label>
  <textarea
    className="form-textarea"
    defaultValue=""
    name="about"
    onChange={[Function]}
  />
</div>
`;

exports[`<TextArea /> when props.defaultValue is valid renders correctly 1`] = `
<div
  className="form-group"
>
  <label>
    About
  </label>
  <textarea
    className="form-textarea"
    defaultValue="lorem ipsum"
    name="about"
    onChange={[Function]}
  />
</div>
`;
